ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/Cluster/ClusterPageVulnerabilities.tsx

```diff
@@ -82,17 +84,28 @@ function ClusterPageVulnerabilities({ clusterId }: ClusterPageVulnerabilitiesPro
                             <Flex alignItems={{ default: 'alignItemsCenter' }}>
                                 <Title headingLevel="h2" className="pf-v5-u-w-50">
                                     {data ? (
-                                        `${pluralize(
-                                            data.cluster.clusterVulnerabilityCount,
-                                            'result'
-                                        )} found`
+                                        `${pluralize(clusterVulnerabilityCount, 'result')} found`
                                     ) : (
                                         <Skeleton screenreaderText="Loading cluster vulnerability count" />
                                     )}
                                 </Title>
                                 {isFiltered && <DynamicTableLabel />}
                             </Flex>
                         </SplitItem>
+                        <SplitItem>
+                            <Pagination
+                                itemCount={clusterVulnerabilityCount}
+                                perPage={perPage}
+                                page={page}
+                                onSetPage={(_, newPage) => setPage(newPage)}
+                                onPerPageSelect={(_, newPerPage) => {
+                                    if (clusterVulnerabilityCount < (page - 1) * newPerPage) {
+                                        setPage(1);
+                                    }
+                                    setPerPage(newPerPage);
+                                }}
+                            />
+                        </SplitItem>
```

Comment:
I noticed this structure for pagination is the same in a few places and requires a little tweaking to adjust for different use cases (changing the count variable for `itemCount` and the logic within `onPerPageSelect`). Since we use all the variables exported from `useURLPagination`, what do you think about simplifying things by creating a wrapper `Pagination` component that takes the count and pagination object?

```
const pagination = useURLPagination(DEFAULT_VM_PAGE_SIZE);
...
return (
    ...
    <CustomPagination itemCount={clusterVulnerabilityCount} pagination={pagination} />
    ...
)
```

We can pass the appropriate values as props to the PF pagination component within the custom pagination component. Maybe that would make things simpler to write out?

Just brainstorming again 🤔

Reply:
Yeah this is a good idea. I've noticed this too and it has been bothering me that this is pretty much copy/pasted throughout a lot of the app. (Although I bet there are quite a few places where this edge case isn't even handled.)

I did recently add a helper function for computing the pagination offset since that logic is duplicated similarly in many places:
```ts
export function getPaginationParams(
    page: number,
    perPage: number
): { offset: number; limit: number } {
    return {
        offset: (page - 1) * perPage,
        limit: perPage,
    };
}
```

We should see how this looks using either:
- A custom wrapped `<Pagination />` component
- A helper function that accepts the top level `pagination` object we get from the hook and the `itemCount`